pkg/admission/init.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"reflect"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+type PluginInitializer interface {
+	Initialize(plugins []Interface)
```

Comment:
> I would prefer to initialize all informers at one place as the PluginInitializer is going to be called more than once at various places. @deads2k what do you think? use the initPluginInformers with CreateSharedNamespaceIndexInformer. Later, extend it with other informers as they come?

I want an interface that takes an `obj` and returns back to me a sharedindexinformer for it.  Same interface would also have methods for getting indexes and the lister types.  Then that interface could be provided via a want.

I'd expect to build that interface separately, ideally based on a dynamic client (which isn't dynamic today :() and a list of groupresources.

This is all internal, so I'm willing to take something close for now and we'll tweak later.